tool/format_ignores.py

```diff
@@ -0,0 +1,226 @@
+"""
+
+import argparse
+import os
+import re
+import sys
+from collections.abc import Iterable, Sequence
+from pathlib import Path
+
+
+def _sort_ignore_list(ignore_text: str, /) -> str:
+    """
+    Sort the items in a pyright ignore list and ensure proper spacing.
+
+    Parameters
+    ----------
+    ignore_text : str
+        Text inside the ignore brackets
+
+    Returns
+    -------
+    str
+        Sorted and formatted ignore list
+    """
+    # Extract items from the ignore list
+    items = ignore_text.strip("[]").split(",")
+
+    # Clean up each item and sort
+    cleaned_items = [item.strip() for item in items if item.strip()]
+    sorted_items = sorted(cleaned_items)
+
+    # Join with comma and space
+    return ", ".join(sorted_items)
+
+
+def _process_file(
+    file_path: str | os.PathLike[str],
+    /,
+    *,
+    check_only: bool = False,
+) -> bool:
+    """
+    Process a file to sort pyright ignore comments.
+
+    Parameters
+    ----------
+    file_path : str or os.PathLike[str]
+        Path to the file to process
+    check_only : bool, default=False
+        If True, don't modify the file, just check if changes are needed
+
+    Returns
+    -------
+    bool
+        True if the file was modified or would be modified (in check mode)
+    """
+    path_obj = Path(file_path)
+    content = path_obj.read_text(encoding="utf-8")
+
+    # Find all pyright ignore comments and sort them
+    pattern = r"(# pyright: ignore\[)([^\]]+)(\])"
+
+    def replace_match(match: re.Match[str], /) -> str:
+        prefix = match.group(1)
+        ignore_list = match.group(2)
+        suffix = match.group(3)
+        sorted_ignore = _sort_ignore_list(ignore_list)
+        return f"{prefix}{sorted_ignore}{suffix}"
+
+    updated_content = re.sub(pattern, replace_match, content)
+
+    # Check if changes would be made
+    needs_changes = content != updated_content
+
+    # Write the updated content back if changes were made and not in check mode
+    if needs_changes and not check_only:
+        path_obj.write_text(updated_content, encoding="utf-8")
+        print(f"Updated {file_path}")
+
+    return needs_changes
+
+
+def _process_directory(
+    directory_path: str | os.PathLike[str],
+    /,
+    *,
+    extensions: Iterable[str] = (".pyi",),
+    check_only: bool = False,
+) -> tuple[int, int]:
+    """
+    Process all files with the given extensions in a directory and its subdirectories.
+
+    Parameters
+    ----------
+    directory_path : str or os.PathLike[str]
+        The path to the directory to process
+    extensions : iterable of str, default=('.pyi',)
+        File extensions to process
+    check_only : bool, default=False
+        If True, don't modify files, just check them
+
+    Returns
+    -------
+    tuple of (int, int)
+        (count of files that would be modified, total files checked)
+    """
+    modified_count = 0
+    total_count = 0
+
+    dir_path = Path(str(directory_path))
```

Comment:
is the `str` needed here?

```suggestion
    dir_path = Path(directory_path)
```